x/evm/types/state_transition.go

```diff
@@ -16,30 +17,58 @@ import (
 
 // StateTransition defines data to transitionDB in evm
 type StateTransition struct {
-	Payload      []byte
-	Recipient    *common.Address
+	// TxData fields
 	AccountNonce uint64
-	GasLimit     uint64
 	Price        *big.Int
+	GasLimit     uint64
+	Recipient    *common.Address
 	Amount       *big.Int
-	ChainID      *big.Int
-	Csdb         *CommitStateDB
-	THash        *common.Hash
-	Sender       common.Address
-	Simulate     bool
+	Payload      []byte
+
+	ChainID  *big.Int
+	Csdb     *CommitStateDB
+	TxHash   *common.Hash
+	Sender   common.Address
+	Simulate bool
+}
+
+// GasInfo returns the gas limit, gas consumed and gas refunded from the EVM transition
+// execution
+type GasInfo struct {
+	GasLimit    uint64
+	GasConsumed uint64
+	GasRefunded uint64
 }
 
-// ReturnData represents what's returned from a transition
-type ReturnData struct {
```

Comment:
Renamed to match geth's:

https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/core/state_transition.go#L71-L77